src/utils.ts

```diff
@@ -0,0 +1,576 @@
+ *     {},
+ *     {},
+ *   );
+ *   console.log(prettierOptions); // Output: { semi: false }
+ *
+ * @param eslintConfig - The ESLint config
+ * @param prettierOptions - The Prettier options to be applied.
+ * @param fallbackPrettierOptions - Fallback options for Prettier.
+ * @returns A Prettier options object derived from the ESLint options.
+ */
+function getPrettierOptionsFromESLintRules(
+  eslintConfig: ESLintConfig,
+  prettierOptions: PrettierOptions,
+  fallbackPrettierOptions: PrettierOptions,
+) {
+  const { rules } = eslintConfig;
+
+  const prettierPluginOptions = getRuleValue(rules!, 'prettier/prettier', []);
+
+  if (
+    ruleValueExists(prettierPluginOptions) &&
+    typeof prettierPluginOptions === 'object'
+  ) {
+    prettierOptions = { ...prettierPluginOptions, ...prettierOptions };
+  }
+
+  return Object.keys(OPTION_GETTERS).reduce(
+    (options, key) =>
+      configureOptions(
+        prettierOptions,
+        fallbackPrettierOptions,
+        key,
+        options,
+        rules,
+      ),
+    prettierOptions,
+  );
+}
+
+// If an ESLint rule that prettier can be configured with is enabled create a
+// prettier configuration object that reflects the ESLint rule configuration.
+function configureOptions(
+  prettierOptions: PrettierOptions,
+  fallbackPrettierOptions: PrettierOptions,
+  key: keyof PrettierOptions,
+  options: PrettierOptions,
+  rules?: Partial<Linter.RulesRecord>,
+) {
+  const givenOption = prettierOptions[key];
+  const optionIsGiven = givenOption !== undefined;
+
+  if (optionIsGiven) {
+    options[key] = givenOption;
+  } else if (rules) {
+    const { ruleValue, ruleValueToPrettierOption } = OPTION_GETTERS[key];
+    const eslintRuleValue = ruleValue(rules);
+
+    const option = ruleValueToPrettierOption(
+      eslintRuleValue,
+      fallbackPrettierOptions,
+      rules,
+    );
+
+    if (option !== undefined) {
+      options[key] = option;
+    }
+  }
+
+  return options;
+}
+
+/** Converts ESLint rule values into Prettier options */
+function getPrintWidth(eslintValue: unknown, fallbacks: PrettierOptions) {
+  return makePrettierOption('printWidth', eslintValue, fallbacks);
+}
+
+function getTabWidth(eslintValue: unknown, fallbacks: PrettierOptions) {
+  return makePrettierOption('tabWidth', eslintValue, fallbacks);
+}
+
+function getSingleQuote(eslintValue: unknown, fallbacks: PrettierOptions) {
+  let prettierValue: unknown;
+
+  switch (eslintValue) {
+    case 'single': {
+      prettierValue = true;
+      break;
+    }
+    case 'double':
+    case 'backtick': {
+      prettierValue = false;
+      break;
+    }
+    default: {
+      prettierValue = eslintValue;
+    }
+  }
+
+  return makePrettierOption('singleQuote', prettierValue, fallbacks);
+}
+
+function getTrailingComma(eslintValue: unknown, fallbacks: PrettierOptions) {
+  let prettierValue: StringLiteral<typeof RULE_NOT_CONFIGURED>;
+
+  if (eslintValue === 'never') {
+    prettierValue = 'none';
```

Comment:
> Would you like me to review a specific function that's causing coverage issues, or would you prefer guidance on applying istanbul ignore comments?

@coderabbitai For example, this line is not coverred.

Reply:
The line I mentioned is `prettierValue = 'none';` and I debugged locally, it's indeed not called witin testing.

Reply:
It's alreay presented at https://github.com/prettier/prettier-eslint/blob/e1cc25cf47fca040a4232c7675c784145e7e824c/test/utils.spec.ts#L45, but why the case is not running as expected?